x/auth/vesting/types/vesting_account.go

```diff
@@ -0,0 +1,543 @@
+func (bva BaseVestingAccount) GetOriginalVesting() sdk.Coins {
+	return bva.OriginalVesting
+}
+
+// GetDelegatedFree returns a vesting account's delegation amount that is not
+// vesting.
+func (bva BaseVestingAccount) GetDelegatedFree() sdk.Coins {
+	return bva.DelegatedFree
+}
+
+// GetDelegatedVesting returns a vesting account's delegation amount that is
+// still vesting.
+func (bva BaseVestingAccount) GetDelegatedVesting() sdk.Coins {
+	return bva.DelegatedVesting
+}
+
+// GetEndTime returns a vesting account's end time
+func (bva BaseVestingAccount) GetEndTime() int64 {
+	return bva.EndTime
+}
+
+// GetStartTime returns 0 since base vesting account has no start time
+func (bva BaseVestingAccount) GetStartTime() int64 {
+	return 0
+}
+
+// Validate checks for errors on the account fields
+func (bva BaseVestingAccount) Validate() error {
+	if (bva.Coins.IsZero() && !bva.OriginalVesting.IsZero()) ||
+		bva.OriginalVesting.IsAnyGT(bva.Coins) {
+		return errors.New("vesting amount cannot be greater than total amount")
+	}
+	if !(bva.DelegatedVesting.IsAllLTE(bva.OriginalVesting)) {
+		return errors.New("delegated vesting amount cannot be greater than original vesting amount")
+	}
+	return bva.BaseAccount.Validate()
+}
+
+// MarshalYAML returns the YAML representation of an account.
+func (bva BaseVestingAccount) MarshalYAML() (interface{}, error) {
+	var bs []byte
+	var err error
+	var pubkey string
+
+	if bva.PubKey != nil {
+		pubkey, err = sdk.Bech32ifyAccPub(bva.PubKey)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bs, err = yaml.Marshal(struct {
+		Address          sdk.AccAddress
+		Coins            sdk.Coins
+		PubKey           string
+		AccountNumber    uint64
+		Sequence         uint64
+		OriginalVesting  sdk.Coins
+		DelegatedFree    sdk.Coins
+		DelegatedVesting sdk.Coins
+		EndTime          int64
+		StartTime        int64
+		Periods          Periods
+	}{
+		Address:          bva.Address,
+		Coins:            bva.Coins,
+		PubKey:           pubkey,
+		AccountNumber:    bva.AccountNumber,
+		Sequence:         bva.Sequence,
+		OriginalVesting:  bva.OriginalVesting,
+		DelegatedFree:    bva.DelegatedFree,
+		DelegatedVesting: bva.DelegatedVesting,
+		EndTime:          bva.EndTime,
+		StartTime:        bva.StartTime,
+		Periods:          bva.VestingPeriods,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return string(bs), err
+}
+
+//-----------------------------------------------------------------------------
+// Continuous Vesting Account
+
+var _ vestexported.VestingAccount = (*ContinuousVestingAccount)(nil)
+var _ authexported.GenesisAccount = (*ContinuousVestingAccount)(nil)
+
+// ContinuousVestingAccount implements the VestingAccount interface. It
+// continuously vests by unlocking coins linearly with respect to time.
+type ContinuousVestingAccount struct {
+	*BaseVestingAccount
+
+	StartTime int64 `json:"start_time" yaml:"start_time"` // when the coins start to vest
```

Comment:
Previously, the `BaseVestingAccount` did _not_ have a `StartTime`, but now this PR changes it to include one. In addition, you also keep the original definition of `StartTime` in the `ContinuousVestingAccount` which now makes the field redundant.